[Quek Xian Yi Erwin] iP

erwinqxy opened the pull request:
# Duke Bot :robot:
As [Elon Musk](https://en.wikipedia.org/wiki/Elon_Musk) once said:
>“You should take the approach that you’re wrong. Your goal is to be less wrong.”

_Target Users_ 
* Teachers 
* Students 

## Features 
**Type of Tasks** 
1. `Todo` 
1. `Event` 
1. `Deadline` 

*Methods** 
1. `markDone` 
1. `getTaskList` 
1. `deleteTask` 

  ## Todo List ~~procastinate~~
- [x] [Week 2 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w2.html) 
- [x] [Week 3 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w3.html) 
- [x] [Week 4 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w4.html) 
- [x] [Week 5 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w5.html) 
- [x] [Week 6 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w6.html) 
- [ ] [Week 7 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w7.html) 
- [ ] [Week 8 Tasks](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w8.html) 

## To run 
To `<run>` the bot, look at the main function 
```java
public static void main(String[] args) throws FileNotFoundException {
        new Duke("data/duke.txt").run();
    }
```

Commit:
Add Gradle support

erwinqxy committed:
Level 1.01 - modified starting message

erwinqxy committed:
Level-1 Completed

erwinqxy committed:
Level-2 Completed

erwinqxy committed:
Complete Level-3

erwinqxy committed:
Push to check tags

erwinqxy committed:
Encapsulate Duke in functions

erwinqxy committed:
added todo, events, deadline

erwinqxy committed:
added Automated Text UI Testing

erwinqxy committed:
handle errors todo and invalid inputs

erwinqxy committed:
updated input, expected

erwinqxy committed:
added delete

erwinqxy committed:
Encapsulate Bye Message

erwinqxy committed:
Implement load local file data

erwinqxy committed:
Add writing file func for todo, event and deadline

erwinqxy committed:
Complete Level-7

erwinqxy committed:
Add datetime attributes for Deadlines and Events

erwinqxy committed:
Merge

erwinqxy committed:
Troublshoot merge conflict

erwinqxy committed:
Merge branch 'branch-Level-8'

erwinqxy committed:
Fix datetime input error in Duke.txt

erwinqxy committed:
Add new Storage class

erwinqxy committed:
Create Ui, Parser, Storage, TaskList classes

erwinqxy committed:
Create duke package

erwinqxy committed:
Add javadocs for Deadline

erwinqxy committed:
Add Javadocs for Deadline

erwinqxy committed:
Add javadocs for Duke

erwinqxy committed:
Add JavaDocs for DukeException

erwinqxy committed:
Add JavaDoc for Event class

erwinqxy committed:
Added Javadocs to ToDo class

erwinqxy committed:
Add JavaDocs for Parser

erwinqxy committed:
Create a JAR File

erwinqxy committed:
Add JUnit Tests

erwinqxy committed:
Follow the Coding Standard

erwinqxy committed:
Revert "Follow the Coding Standard"

This reverts commit 338cdf55912a508ef08ab11e875318e112973723.

erwinqxy committed:
Update DukeTest.java

erwinqxy committed:
Added Find method

erwinqxy committed:
Add JavaDocs for Storage, Task, TaskList and Ui classes

erwinqxy committed:
Merge branch 'branch-Level-9'

erwinqxy committed:
Merge branch 'branch-A-JavaDoc'

erwinqxy committed:
Update Coding Standards

erwinqxy committed:
Update duke.txt

erwinqxy committed:
Merge branch 'branch-A-CodingStandard'

erwinqxy committed:
Add Javadocs for Duke.java

AllardQuek commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+
+    /**
+     * Constructor for Deadline.
+     * @param description Takes in a description of Deadline task
```
A description doesn't take in a description

erwinqxy replied:
ok noted! I will make the necessary changes. Thank you 😄

AllardQuek commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * A method to get deadline task info.
```
Use "Returns" instead

erwinqxy replied:
Thank you for that feedback 😄 ! i will update it

AllardQuek commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+                Scanner s = new Scanner(System.in);
+                String command = s.nextLine();
+                isExit = Parser.parse(command);
+                //Command c = duke.Parser.parse(fullCommand);
+                //c.execute(tasks, ui, storage);
+            //} catch (duke.DukeException e) {
+            //    ui.showError(e.getMessage());
```
Maybe you should remove irrelevant comments?

erwinqxy replied:
I was thinking if i want to work on this later on. but I think I will remove it! Thank you

KengXIII commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Method to mark a task done by index.
```
Words like Returns ..., Sends ..., Adds ... would be a better way to start off the JavaDocs

erwinqxy replied:
Thank you for your suggestions. I will keep that in mind and make the relevant changes  😄

KengXIII commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+     */
+    public static void deleteTask(int i) throws DukeException {
+        System.out.println("Noted. I've removed this task: ");
+        String deleted = tasks.getTask(i-1).toString();
```
The variable name 'deleted' could be a little more descriptive here to avoid confusions.

erwinqxy replied:
Oh true! Thank you for that suggestion!

KengXIII commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+        }
+    }
+
+//    public TaskList() {
```
Old commented-out code that could be removed

erwinqxy replied:
I was contemplating if I should work on it further. but I think I will remove it! Thank you for reminding me~ 😄

KengXIII reviewed (commented):
I think the code generally looks great. However, the JavaDocs description may need some refining to adhere to the coding conventions. Good job!

erwinqxy committed:
Merge branch 'add-gradle-support'

erwinqxy committed:
Add a GUI to Duke

AllardQuek commented on data/duke.txt:
```diff
@@ -5,7 +5,4 @@ T|1|join sports club
-T|0|sleep
-T|0|sleep
-T|0|slap allard
-T|0|slap allard 3 times
```
I like that you removed such offensive tasks.

AllardQuek commented on src/main/java/duke/tools/Ui.java:
```diff
@@ -55,24 +55,28 @@ public String testReadCommand() {
+    }
 }
 
 //Code to expand on in the future
```
Do you think there might be a better way to manage your work in progress code besides committing them to your git history? 🙂

erwinqxy replied:
any suggestions? another branch perhaps

AllardQuek replied:
try this link: jibble.org/stfw

erwinqxy committed:
Fix spacing issues

erwinqxy committed:
Fix spacing issues

erwinqxy committed:
Fix spacing

erwinqxy committed:
Fix spacing

erwinqxy committed:
Add error handling

erwinqxy committed:
Add error handling

erwinqxy committed:
Update Ui.java

Remove irrelevant code.

To improve on quality of code.

erwinqxy committed:
Update Parser.java

Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code in Parser

erwinqxy committed:
Fix code quality for tasks

Clean up the code quality of tasks.
To improve readability.

erwinqxy committed:
Fix code quality

erwinqxy committed:
Merge pull request #1 from erwinqxy/branch-A-Assertions

Update Parser.java

erwinqxy committed:
Merge branch 'master' into branch-A-CodeQuality

erwinqxy committed:
Merge pull request #2 from erwinqxy/branch-A-CodeQuality

Branch A-CodeQuality

erwinqxy committed:
Add start up message for GUI

erwinqxy committed:
Update DialogBox.fxml

erwinqxy committed:
Update image

erwinqxy committed:
Add C-DetectDuplicates

erwinqxy committed:
Update TaskList.java

erwinqxy committed:
Update duke.txt

erwinqxy committed:
Add checkstyle

erwinqxy committed:
Update TaskList.java

erwinqxy committed:
Merge branch 'branch-C-DetectDuplicates'

erwinqxy committed:
Improve GUI

erwinqxy committed:
Update GUI

erwinqxy committed:
Update GUI

erwinqxy committed:
Update DialogBox.fxml

erwinqxy committed:
Add representative product screenshot

erwinqxy committed:
Set theme jekyll-theme-midnight

erwinqxy committed:
Set theme jekyll-theme-cayman

erwinqxy committed:
Set theme jekyll-theme-midnight

erwinqxy committed:
Update UI.png

erwinqxy committed:
Update Parser.java

erwinqxy committed:
Merge branch 'master' of https://github.com/erwinqxy/ip

erwinqxy committed:
Add create duke.txt

erwinqxy committed:
Fix CLI message output

erwinqxy committed:
Add terminate Main Window

erwinqxy committed:
Fix javafx fml version

erwinqxy committed:
Update README.md

erwinqxy committed:
Set theme jekyll-theme-hacker

erwinqxy committed:
Update README.md

erwinqxy committed:
Merge branch 'master' of https://github.com/erwinqxy/ip

erwinqxy committed:
Update DialogBox.fxml

erwinqxy committed:
Update javadocs

erwinqxy committed:
Update Javadocs

erwinqxy committed:
Update Javadocs

erwinqxy committed:
Add more Unit Test

erwinqxy committed:
Update README.md

erwinqxy committed:
Update README.md

erwinqxy committed:
Update new image files

erwinqxy committed:
Update Error Message

erwinqxy committed:
Update README.md

erwinqxy committed:
Add icon to Main Window

erwinqxy committed:
Update coding standards

erwinqxy committed:
Update Ui.png
